cdap-master/src/main/java/io/cdap/cdap/master/environment/k8s/RouterServiceMain.java

```diff
@@ -61,81 +55,41 @@ public static void main(String[] args) throws Exception {
   }
 
   @Override
-  protected List<Module> getServiceModules(
-      MasterEnvironment masterEnv,
-      EnvironmentOptions options, CConfiguration cConf) {
-    return Arrays.asList(
-        new MessagingClientModule(),
-        new RouterModules().getDistributedModules(),
-        new DFSLocationModule(),
-        new ZKClientModule(),
-        securityModuleBasedOn(cConf)
-    );
+  protected List<Module> getServiceModules(MasterEnvironment masterEnv,
+                                           EnvironmentOptions options, CConfiguration cConf) {
+    List<Module> modules = new ArrayList<>();
+    modules.add(new MessagingClientModule());
+    modules.add(new RouterModules().getDistributedModules());
+    modules.add(new DFSLocationModule());
+    modules.addAll(getSecurityModules(cConf));
+
+    return modules;
   }
 
-  private Module securityModuleBasedOn(CConfiguration cConf) {
-    if (cConf.getBoolean(Constants.Security.ENABLED)) {
-      return new SecurityModules().getDistributedModules();
-    } else {
-      return new SecurityModules().getStandaloneModules();
+  private List<Module> getSecurityModules(CConfiguration cConf) {
+    if (!cConf.getBoolean(Constants.Security.ENABLED)) {
```

Comment:
Is this for backwards-compatible behavior, i.e. is the KeyManager used even when security isn't enabled?

Reply:
Some of the bindings are needed for injection to provide no-op / unsupported exception

Reply:
Acknowledged, thanks!